src/pudl/helpers.py

```diff
@@ -1513,7 +1513,9 @@ def weighted_average(
         A table with ``by`` columns as the index and the weighted ``data_col``.
     """
     df["_data_times_weight"] = df[data_col] * df[weight_col]
-    df["_weight_where_notnull"] = df.loc[df[data_col].notnull(), weight_col]
+    df.loc[df[data_col].notnull(), "_weight_where_notnull"] = df.loc[
+        df[data_col].notnull(), weight_col
+    ]
```

Comment:
I'm very confused why you and the builds would see this error, but I wouldn't. It was fine on the parquet branch build I ran last night. 🤷🏼 

I think there's a built in way to do this using the built-in [`df.where()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.where.html) if I understand the desired outcome. Does this seem correct?

```suggestion
df["_weight_where_notnull"] = df[weight_col].where(df[data_col].notnull())]
```

Reply:
i'm confused why this was a problem all of a sudden at all.

but that suggestion looks fine. you have an extra `]` in there it should be:

```python
df["_weight_where_notnull"] = df[weight_col].where(df[data_col].notnull())
```

Reply:
Okay fixed.